trino/auth.py

```diff
@@ -128,11 +128,18 @@ def get_exceptions(self):
     def handle_error(self, handle_error):
         pass
 
+    def __eq__(self, other):
```

Comment:
Do all `Authentication` impls need to implement this? If so, let's move this to `Authentication` with a default that `raise`s an exception maybe?

Reply:
Like java, `__eq__` available in all object, but by default, it check object identity.   
I added `__eq__` to all `Authentication` impls, but I don't think raise an exception in base `Authentication` class is good idea, because it's public API and may introduce breaking change if some one has they own `Authentication`